pkg/csi/service/wcp/controller.go

```diff
@@ -749,8 +749,8 @@ func (c *controller) createFileVolume(ctx context.Context, req *csi.CreateVolume
 	}
 
 	if len(filteredDatastores) == 0 {
-		return nil, csifault.CSIInternalFault, logger.LogNewErrorCode(log, codes.Internal,
-			"no datastores found to create file volume")
+		return nil, csifault.CSIVSanFileServiceDisabled, logger.LogNewErrorCode(log, codes.Internal,
```

Comment:
Can we call IsvSANFileServiceEnabled here when `len(filteredDatastores) == 0` to determine if you can return `CSIVSanFileServiceDisabled` or `CSIInternalFault`?

Reply:
I read the code again, when len(filteredDatastores) == 0, it only means file service is not enabled on any cluster in that vc. So we can return  "CSIVSanFileServiceDisabled" here. I have modified the code to simplify the logic.